packages/client/src/components/ui/pagination.vue

```diff
@@ -62,24 +74,72 @@ const props = withDefaults(defineProps<{
 });
 
 const emit = defineEmits<{
-	(e: 'queue', count: number): void;
+	(ev: 'queue', count: number): void;
 }>();
 
-type Item = { id: string; [another: string]: unknown; };
+let rootEl = $ref<HTMLElement>();
 
-const rootEl = ref<HTMLElement>();
-const items = ref<Item[]>([]);
-const queue = ref<Item[]>([]);
+/*
+ * 遡り中かどうか
+ */
+let backed = $ref(false);
+
+let scrollRemove: (() => void) | null = $ref(null);
+
+const items = ref<MisskeyEntity[]>([]);
+const queue = ref<MisskeyEntity[]>([]);
 const offset = ref(0);
 const fetching = ref(true);
 const moreFetching = ref(false);
 const more = ref(false);
-const backed = ref(false); // 遡り中か否か
 const isBackTop = ref(false);
 const empty = computed(() => items.value.length === 0);
 const error = ref(false);
+const {
+	enableInfiniteScroll
+} = defaultStore.reactiveState;
+
+const contentEl = $computed(() => props.pagination.pageEl || rootEl);
+const scrollableElement = $computed(() => getScrollContainer(contentEl));
+
+// 先頭が表示されているかどうかを検出
+// https://qiita.com/mkataigi/items/0154aefd2223ce23398e
+const scrollObserver = $computed(() => new IntersectionObserver(entries => {
+	backed = entries[0].isIntersecting;
+}, {
+	root: scrollableElement,
+	rootMargin: props.pagination.reversed ? "-100% 0px 100% 0px" : "100% 0px -100% 0px",
+	threshold: 0.01,
+}));
+
+watch($$(rootEl), () => {
```

Comment:
Since `roolEl` is already a ref, I don't think `$$` is needed here?
```suggestion
watch(rootEl, () => {
```